cmd/fastqFilter/fastqFilter.go

```diff
@@ -64,6 +92,21 @@ func fastqFilter(s Settings) {
 					continue
 				}
 			}
+			if s.RetainNamesList != "" {
+				if _, FwdInMap = namesMap[i.Fwd.Name]; !FwdInMap {
+					if _, RevInMap = namesMap[i.Rev.Name]; !RevInMap {
+						continue //continue if neither the forward or reverse read is in the map
+					}
+				}
+			}
+			if s.DiscardNamesList != "" {
+				if _, FwdInMap = namesMap[i.Fwd.Name]; FwdInMap {
+					continue
+				}
+				if _, RevInMap = namesMap[i.Rev.Name]; !RevInMap {
```

Comment:
I was expecting "RevInMap" instead of "!RevInMap" towards the end of the line.

Reply:
Great catch! Copy paste error. Fixed in next commit